# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../mc-mp/stropts.patch
# Copyright (C) 2007 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

uClibC and dietlibc do not have those.

  - Rene Rebe <rene@exactcode.de>

--- mc-4.1.40-pre9/src/subshell.c.vanilla	2007-01-15 09:36:52.000000000 +0000
+++ mc-4.1.40-pre9/src/subshell.c	2007-01-15 09:42:57.000000000 +0000
@@ -76,7 +76,14 @@
 #endif
 
 #ifdef HAVE_GRANTPT
+#if defined (__UCLIBC__) || defined (__dietlibc__)
+#define __SID           ('S' << 8)
+#define I_PUSH      (__SID | 2)
+#define I_POP       (__SID | 3)
+#define I_FIND      (__SID |11)
+#else
 #   include <stropts.h> /* For I_PUSH			      */
+#endif
 #else
 #   include <grp.h>	/* For the group struct & getgrnam()  */
 #endif
